server/src/main/java/org/opensearch/search/aggregations/bucket/terms/MultiTermsAggregator.java

```diff
@@ -0,0 +1,435 @@
+        this.bucketOrds = BytesKeyedBucketOrds.build(context.bigArrays(), cardinality);
+        this.multiTermsValue = new MultiTermsValuesSource(internalValuesSources);
+        this.showTermDocCountError = showTermDocCountError;
+        this.formats = formats;
+        this.bucketCountThresholds = bucketCountThresholds;
+        this.order = order;
+        this.partiallyBuiltBucketComparator = order == null ? null : order.partiallyBuiltBucketComparator(b -> b.bucketOrd, this);
+        // Todo, copy from TermsAggregator. need to remove duplicate code.
+        if (subAggsNeedScore() && descendsFromNestedAggregator(parent)) {
+            /**
+             * Force the execution to depth_first because we need to access the score of
+             * nested documents in a sub-aggregation and we are not able to generate this score
+             * while replaying deferred documents.
+             */
+            this.collectMode = SubAggCollectionMode.DEPTH_FIRST;
+        } else {
+            this.collectMode = collectMode;
+        }
+        // Don't defer any child agg if we are dependent on it for pruning results
+        if (order instanceof InternalOrder.Aggregation) {
+            AggregationPath path = ((InternalOrder.Aggregation) order).path();
+            aggsUsedForSorting.add(path.resolveTopmostAggregator(this));
+        } else if (order instanceof InternalOrder.CompoundOrder) {
+            InternalOrder.CompoundOrder compoundOrder = (InternalOrder.CompoundOrder) order;
+            for (BucketOrder orderElement : compoundOrder.orderElements()) {
+                if (orderElement instanceof InternalOrder.Aggregation) {
+                    AggregationPath path = ((InternalOrder.Aggregation) orderElement).path();
+                    aggsUsedForSorting.add(path.resolveTopmostAggregator(this));
+                }
+            }
+        }
+    }
+
+    @Override
+    public InternalAggregation[] buildAggregations(long[] owningBucketOrds) throws IOException {
+        InternalMultiTerms.Bucket[][] topBucketsPerOrd = new InternalMultiTerms.Bucket[owningBucketOrds.length][];
+        long[] otherDocCounts = new long[owningBucketOrds.length];
+        for (int ordIdx = 0; ordIdx < owningBucketOrds.length; ordIdx++) {
+            collectZeroDocEntriesIfNeeded(owningBucketOrds[ordIdx]);
+            long bucketsInOrd = bucketOrds.bucketsInOrd(owningBucketOrds[ordIdx]);
+
+            int size = (int) Math.min(bucketsInOrd, bucketCountThresholds.getShardSize());
+            PriorityQueue<InternalMultiTerms.Bucket> ordered = new BucketPriorityQueue<>(size, partiallyBuiltBucketComparator);
+            InternalMultiTerms.Bucket spare = null;
+            BytesRef dest = null;
+            BytesKeyedBucketOrds.BucketOrdsEnum ordsEnum = bucketOrds.ordsEnum(owningBucketOrds[ordIdx]);
+            CheckedSupplier<InternalMultiTerms.Bucket, IOException> emptyBucketBuilder = () -> InternalMultiTerms.Bucket.EMPTY(formats);
```

Comment:
Should the empty bucket also include `showDocCountError` setting? `InternalMultiTerms.Bucket.EMPTY(showDocCountError, formats);`

Reply:
Done.